core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala

```diff
@@ -79,12 +79,17 @@ class JdbcRDD[T: ClassTag](
     val conn = getConnection()
     val stmt = conn.prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)
 
-    // setFetchSize(Integer.MIN_VALUE) is a mysql driver specific way to force streaming results,
-    // rather than pulling entire resultset into memory.
-    // see http://dev.mysql.com/doc/refman/5.0/en/connector-j-reference-implementation-notes.html
-    if (conn.getMetaData.getURL.matches("jdbc:mysql:.*")) {
+    val url = conn.getMetaData.getURL
+    if (url.startsWith("jdbc:mysql:")) {
+      // setFetchSize(Integer.MIN_VALUE) is a mysql driver specific way to force streaming results,
+      // rather than pulling entire resultset into memory.
+      // see https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-reference-implementation-notes.html
+
       stmt.setFetchSize(Integer.MIN_VALUE)
-      logInfo("statement fetch size set to: " + stmt.getFetchSize + " to force MySQL streaming ")
+      logInfo("statement fetch size set to: " + stmt.getFetchSize + " to force MySQL streaming")
+    } else {
+      stmt.setFetchSize(100)
+      logInfo("statement fetch size set to: " + stmt.getFetchSize + " to force streaming")
```

Comment:
Nit: you could use string interpolation in both statements. Also, it's not really streamed a record at a time in this case here. Maybe just log the statement fetch size.